src/overlays/actors/ovl_En_Talk/z_en_talk.c

```diff
@@ -30,14 +28,44 @@ const ActorInit En_Talk_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void EnTalk_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTalk* this = THIS;
+    s8 targetMode;
+
+    targetMode = this->actor.home.rot.x - 0x1;
+
+    if (targetMode >= 0x0 && targetMode < 0x7) {
+        this->actor.targetMode = targetMode;
+    }
+
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = func_80BDE090;
+    this->actor.textId = GET_TEXT_ID(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk/EnTalk_Init.s")
+void EnTalk_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTalk* this = THIS;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk/EnTalk_Destroy.s")
+void func_80BDE058(EnTalk* this, GlobalContext* globalCtx) {
+    if (func_800B867C(&this->actor, globalCtx) != 0) {
+        this->actionFunc = func_80BDE090;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk/func_80BDE058.s")
+void func_80BDE090(EnTalk* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        this->actionFunc = func_80BDE058;
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk/func_80BDE090.s")
+    if ((this->actor.xzDistToPlayer < 40.0f && Actor_IsLinkFacingActor(&this->actor, 0x3000, globalCtx)) ||
+        this->actor.isTargeted) {
+        func_800B8614(&this->actor, globalCtx, 120.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk/EnTalk_Update.s")
+void EnTalk_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTalk* this = THIS;
+    this->actionFunc(this, globalCtx);
```

Comment:
```suggestion
    EnTalk* this = THIS;
    
    this->actionFunc(this, globalCtx);
```
Common convention is to have a space after the variable declarations.